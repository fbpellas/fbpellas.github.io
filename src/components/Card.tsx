import * as React from 'react';
import { Card as CardBootstrap } from 'react-bootstrap';
import Button from 'react-bootstrap/Button';

export interface CardProps {
  setPageAndClear: (hash: string) => void;
  description: string;
  href: string;
  title: string;
}

const Card: React.FC<CardProps> = (props) => {
  const { description, href, setPageAndClear, title } = props;

  return (
    <CardBootstrap>
      <CardBootstrap.Body>
        <CardBootstrap.Text>{title}</CardBootstrap.Text>
        <CardBootstrap.Title>{description}</CardBootstrap.Title>
        <Button href={`#${href}`} onClick={() => setPageAndClear(href)} variant="secondary">
          Learn
        </Button>
      </CardBootstrap.Body>
    </CardBootstrap>
  );
};

export { Card };
