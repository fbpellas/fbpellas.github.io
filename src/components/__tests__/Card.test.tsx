import React from 'react';
import { render } from '@testing-library/react';
import { Card } from '../Card';

test('Card', () => {
  const fn = jest.fn();
  const { container } = render(<Card description="desc" href="some-page" title="title" setPageAndClear={fn} />);
  expect(container).toMatchSnapshot();
});
