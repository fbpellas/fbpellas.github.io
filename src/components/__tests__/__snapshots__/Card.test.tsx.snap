// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <p
        class="card-text"
      >
        title
      </p>
      <div
        class="card-title h5"
      >
        desc
      </div>
      <a
        class="btn btn-secondary"
        href="#some-page"
      >
        Learn
      </a>
    </div>
  </div>
</div>
`;
